### PR TITLE
DBZ-2024 Adding topic.regex option to routing and filtering SMTs

### PR DESCRIPTION
**Pull request by gunnarmorling**
https://issues.redhat.com/browse/DBZ-2024

### CODE REVIEW

**Comment by gunnarmorling** [comment]
@Naros, could you review and merge this one? Thx!

File: debezium-core/src/main/java/io/debezium/transforms/ScriptingTransformation.java [hunk 1]
```diff
@@ -93,13 +95,21 @@ public static NullHandling parse(String value, String defaultValue) {
         }
     }
 
+    private static final Field TOPIC_REGEX = Field.create("topic.regex")
```
Comment (jpechane) [inline debezium-core/src/main/java/io/debezium/transforms/ScriptingTransformation.java:98]: @gunnarmorling Could we have a list of regexes?

Comment (gunnarmorling) [reply debezium-core/src/main/java/io/debezium/transforms/ScriptingTransformation.java:98]: What's the advantage over simply using "foo|bar"?

File: documentation/modules/ROOT/pages/configuration/content-based-routing.adoc [hunk 2]
```diff
@@ -92,6 +92,10 @@ value.op == 'u' ? 'updates' : null
 |Default
 |Description
 
+|`topic.regex`
```
Comment (Naros) [inline documentation/modules/ROOT/pages/configuration/content-based-routing.adoc:95]: Should we consider adding the configuration property links like we've began doing for the connectors?

File: documentation/modules/ROOT/pages/configuration/filtering.adoc [hunk 3]
```diff
@@ -96,6 +96,10 @@ value.op == 'u' && value.before.id == 2
 |Default
 |Description
 
+|`topic.regex`
```
Comment (Naros) [inline documentation/modules/ROOT/pages/configuration/filtering.adoc:99]: Should we consider adding the configuration property links like we've began doing for the connectors?

**Comment by gunnarmorling** [comment]
> Should we consider adding the configuration property links like we've
began doing for the connectors?

Ah yes, good thinking. Done for those two pages.

File: debezium-core/src/main/java/io/debezium/transforms/ScriptingTransformation.java (hunk 1 above)
Comment (Naros) [reply debezium-core/src/main/java/io/debezium/transforms/ScriptingTransformation.java:98]: Was there anything else left to discuss on this point?

Comment (gunnarmorling) [reply debezium-core/src/main/java/io/debezium/transforms/ScriptingTransformation.java:98]: I think that not, if you agree that giving multiple patterns with the pipe is good enough. I'd argue it's even better, as comma-separated values will require escaping.

Comment (gunnarmorling) [reply debezium-core/src/main/java/io/debezium/transforms/ScriptingTransformation.java:98]: In other words, feel free to merge it, @Naros :)

**Comment by jpechane** [comment]
@gunnarmorling Applied, thanks

**Pull request merged**